build/wafutils.py

```diff
@@ -141,6 +141,31 @@ def check_c_header_cached(conf, **kw):
     return result
 
 
+def check_c_func_in_headers_cached(conf, **kw):
+    """
+    Check for @function_name in any of @header_name.
+    Use like a conf.check_cc() for a lib, but you can pass a list of
+    headers in @header_name.
+
+    This checks for the headers firs with check_c_header_cached(), then
+    if this succeeded calls conf.check_cc() on it.
+    """
+    headers = kw.get('header_name')
+    mandatory = kw.get('mandatory', True)
+
+    exception = None
+    for header in [headers] if type(headers) is str else headers:
```

Comment:
Since this whole function only makes sense if more than one header name is passed in, I would rename the argument header_name into header_names and so intimate that a sequence has to be passed in.
This would also make the for-loop more easy to read because we don't need to check for input type.

Also the mandatory and header_names arguments could be listed explicitly in the function definition instead of 'hiding' them in **kw.